client/src/components/Gallery.jsx

```diff
@@ -1,29 +1,56 @@
 import React from 'react';
 import Carousel from './Carousel.jsx';
-import {mount, shallow} from 'enzyme';
+
+const testImages = ['https://loremflickr.com/320/240/foods?random=1',
+'https://loremflickr.com/320/240/foods?random=2', 'https://loremflickr.com/320/240/foods?random=3', 'https://loremflickr.com/320/240/foods?random=4', 'https://loremflickr.com/320/240/foods=5', 'https://loremflickr.com/320/240/foods?random=6', 'https://loremflickr.com/320/240/foods?random=7', 'https://loremflickr.com/320/240/foods?random=8', 'https://loremflickr.com/320/240/foods?random=9', 'https://loremflickr.com/320/240/foods?random=10']
 
 class Gallery extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      imageUrl: 'https://loremflickr.com/320/240/foods'
+      photos: testImages,
+      mainPhoto: testImages[0],
+      mainPhotoIndex: 0,
+      carouselBeginIndex: 0, // use later for array of photos
+      carouselEndIndex: 3
     }
     this.carouselLeftArrowClick = this.carouselLeftArrowClick.bind(this);
     this.carouselRightArrowClick = this.carouselRightArrowClick.bind(this);
   }
 
   carouselLeftArrowClick() {
-    console.log('left');
+    // console.log('left');
+    let mainPhotoIndex = this.state.mainPhotoIndex;
+    let firstPhoto = mainPhotoIndex === 0;
+    let index = firstPhoto ? 0 : mainPhotoIndex - 1;
+    this.setState({
+      mainPhotoIndex: index
+    }, () => {
+      this.setState({
+        mainPhoto: testImages[this.state.mainPhotoIndex]
```

Comment:
Maybe consider only setting state once here? so mainPhotoIndex: index, mainPhoto: testImages[index]? Not positive on this, but it looks like you're re-rendering twice this way

Reply:
Thanks fo reviewing Trevor. I'm updating the mainPhoto with the index which is async so I had to put it in a callback. Otherwise, the index for mainPhoto doesn't update right away when the arrow gets clicked.